### ISSUE TITLE
zprint.main raises IOException: Stream closed

### ISSUE THREAD

**Issue by RickMoynihan** [kkinnear/zprint#57]
With the following in my `~/.clojure/deps.edn`:

```clojure
  :zprint {:extra-deps {zprint/zprint {:mvn/version "0.4.9"}}
           :main-opts ["-m" "zprint.main" ]}
```

(Incidentally it might be nice to add instructions on the above to your README)

Running the command:

`cat ~/path/to/any/clj-file/i/tried.clj | clj -A:zprint`

Formats the file to stdout and prints the exception:

```
Exception in thread "main" java.io.IOException: Stream closed
	at sun.nio.cs.StreamEncoder.ensureOpen(StreamEncoder.java:45)
	at sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:140)
	at java.io.OutputStreamWriter.flush(OutputStreamWriter.java:229)
	at clojure.core$flush.invokeStatic(core.clj:3703)
	at clojure.main$main.invokeStatic(main.clj:424)
	at clojure.main$main.doInvoke(main.clj:387)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:702)
	at clojure.main.main(main.java:37)
```

**Comment by kkinnear** [comment]
What a nice opportunity to learn a bit about the new Clojure tools!  I have reproduced this, and it is interesting.  I use `spit` to write to `*out*` (stdout), and it looks like `clj` must try to close stdout and it gets an exception because `spit` has already closed stdout.  If I don't close stdout, it works fine.  I'm still thinking on how to fix this, but I think I'll have two mains -- one for using with `clj`, and one for my uberjar, since I think closing stdout is probably a good idea in the uberjar.

You suggested that it might be nice to document how to use `clj` the way that you did it.  I didn't do so because I didn't know how, though now of course I do.  My question for you is -- why would you want to do this, as opposed to using the uberjar directly with `java -jar` or with `appcds` for fast startup?  Is this just that much more convenient, and you don't have to download the uberjar from GitHub and install it with `appcds`?  I'm not against another way us using zprint, but I'm interested in your use-case just so I can understand why you are doing this.

In any case, I expect to fix this in the next release, which I'm hoping will come next week sometime, though it could be a little longer.  If you can ignore the exception, the output is fine -- as you have noticed.

**Comment by kkinnear** [comment]
Fixed in 0.4.10.  I mentioned this in the release documentation, but haven't put it in the readme because I'm still uncertain why someone would prefer this to the appcds approach.
